engine/src/main/java/org/terasology/rendering/dag/nodes/PrePostCompositeNode.java

```diff
@@ -190,6 +154,8 @@ public void process() {
             prePostMaterial.setFloat4("skyInscatteringSettingsFrag", skyInscatteringSettingsFrag, true);
         }
 
+        // TODO: We never set the "fogWorldPosition" uniform in prePostComposite_frag.glsl . Either use it, or remove it.
+
```

Comment:
Fog, both in its volumetric and non volumetric form, is something begla (Terasology founder) was experimenting with and he managed to even make some nice videos about it. 

At some point this feature broke and begla never came back to fix it. I'd say: let's leave all that apparatus in and let's either fix it or remove it when we can dig our teeth in the actual shaders. Right now let's just focus on working on the machinery _around_ the shaders and eventually, on splitting the shaders so that we can have a larger number of simpler nodes. After that, or even during the splitting stage if we must, we can have a look at what to do about the fog.